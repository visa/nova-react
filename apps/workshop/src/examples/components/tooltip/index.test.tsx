/**
 *              Â© 2025 Visa
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *         http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 **/
import { fireEvent, render, screen } from '@testing-library/react';
import { axe } from 'jest-axe';

import metaData from './meta.json';

import { BottomTooltip } from './bottom-tooltip';
import { LeftTooltip } from './left-tooltip';
import { RightTooltip } from './right-tooltip';
import { TopTooltip } from './top-tooltip';

const examples = [
  { Component: BottomTooltip, title: metaData['bottom-tooltip'].title },
  { Component: LeftTooltip, title: metaData['left-tooltip'].title },
  { Component: RightTooltip, title: metaData['right-tooltip'].title },
  { Component: TopTooltip, title: metaData['top-tooltip'].title },
];

describe('Tooltip examples', () => {
  examples.forEach(({ Component, title }) => {
    it(`${title} should render correctly`, async () => {
      const { container } = render(<Component />);
      const results = await axe(container);
      expect(results).toHaveNoViolations();
      expect(container).toMatchSnapshot();
    });
  });

  describe('bottom tooltip', () => {
    it('should show on hover', async () => {
      render(<BottomTooltip />);
      const button = screen.getByText<HTMLButtonElement>('Primary action');
      button.focus();
      fireEvent.mouseOver(button);
      expect(screen.queryByText('This is a tooltip')).not.toBeInTheDocument();
    });
  });

  describe('left tooltip', () => {
    it('should show on hover', async () => {
      render(<LeftTooltip />);
      const button = screen.getByText<HTMLButtonElement>('Primary action');
      button.focus();
      fireEvent.mouseOver(button);
      expect(screen.queryByText('This is a tooltip')).not.toBeInTheDocument();
    });
  });

  describe('right tooltip', () => {
    it('should show on hover', async () => {
      render(<RightTooltip />);
      const button = screen.getByText<HTMLButtonElement>('Primary action');
      button.focus();
      fireEvent.mouseOver(button);
      expect(screen.queryByText('This is a tooltip')).not.toBeInTheDocument();
    });
  });

  describe('top tooltip', () => {
    it('should show on hover', async () => {
      render(<TopTooltip />);
      const button = screen.getByText<HTMLButtonElement>('Primary action');
      button.focus();
      fireEvent.mouseOver(button);
      expect(screen.queryByText('This is a tooltip')).not.toBeInTheDocument();
    });
  });
});
