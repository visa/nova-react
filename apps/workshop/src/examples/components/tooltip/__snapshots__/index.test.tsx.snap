// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip examples Bottom-positioned tooltip should render correctly 1`] = `
<div>
  <div
    class="v-flex v-justify-content-center v-m-24"
  >
    <button
      class="v-button"
    >
      Primary action
    </button>
  </div>
</div>
`;

exports[`Tooltip examples Left-positioned tooltip should render correctly 1`] = `
<div>
  <div
    class="v-flex v-justify-content-center v-m-24"
  >
    <button
      class="v-button"
    >
      Primary action
    </button>
  </div>
</div>
`;

exports[`Tooltip examples Right-positioned tooltip should render correctly 1`] = `
<div>
  <div
    class="v-flex v-justify-content-center v-m-24"
  >
    <button
      class="v-button"
    >
      Primary action
    </button>
  </div>
</div>
`;

exports[`Tooltip examples Top-positioned tooltip should render correctly 1`] = `
<div>
  <div
    class="v-flex v-justify-content-center v-m-24"
  >
    <button
      class="v-button"
    >
      Primary action
    </button>
  </div>
</div>
`;
